### ISSUE TITLE
OpenHAB JRuby Binding should not be in this repo

### ISSUE THREAD

**Issue by boc-tothefuture** [boc-tothefuture/openhab-jruby#93]
The binary for the binding should not be tracked in this repo and should be removed from git and the git history so we don't have large binary objects being tracked and downloaded.

I think when I do this, it will be a force push. 

@jimtng It might cause some issues on your side with any open branches?  So we will want to time it.

**Comment by jimtng** [comment]
Go ahead

**Comment by boc-tothefuture** [comment]
Does anyone know what happens when I force push to the main branch?
@jimtng and @pacive  ?  I am worried it will break any active branches or PR?  Can a rebase happen after that?

**Comment by pacive** [comment]
No idea tbh, but i have my branches saved locally and up to date, so would just have to rebase and resubmit the prs in the worst case.

**Comment by jimtng** [comment]
Should this issue remain open @boc-tothefuture ?

**Comment by boc-tothefuture** [comment]
Yes, as soon as we can clear all open PRs, I will remove the large files we have in the repo.

**Comment by boc-tothefuture** [comment]
[This is the process](https://rtyley.github.io/bfg-repo-cleaner/) to be used (unless someone knows a different one), but once we do it, everyone will have to delete their copy of the repo and pull down a fresh copy, so all in progress branches, etc will be lost or need to be copied over to the new clone.

@jimtng and @ccutrer - I would like to do this soonish - are you comfortable with copying everything you have in open branches to a new mirror?

**Comment by ccutrer** [comment]
It's not strictly necessary to delete our own clones, but will need to `reset --hard` over to the new main branch, then can easily cherry-pick in-progress PRs on top and force-push them. I think you can do it via rebase as well, but not the basic rebase. Tl;dr: I'm fine whenever, and have no problem picking up my work.

**Comment by jimtng** [comment]
Go ahead

**Comment by ccutrer** [comment]
Don't forget to delete or "rebase" your other branches

**Comment by boc-tothefuture** [comment]
It should be done.. @ccutrer can you share what you did to rebase.. did you do a rebase or something else?

**Comment by ccutrer** [comment]
I actually just made a new commit, because I had lots of changes I had "made" in my live openhab instance that I copied/pasted over. But I did have to reopen the PR _before_ I force-pushed a new commit over the branch, otherwise GitHub wouldn't let me re-open afterwards. If I had not done a new commit, it would have been as easy as `git reset --hard origin/main && git cherry-pick <hash of main commit in PR> && git push ccutrer HEAD:semantics -f`.
